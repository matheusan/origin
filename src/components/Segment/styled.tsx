import styled from 'styled-components';

export const Segment: JSX.Element = styled.section`
  max-width: 480px;
  margin: 0 auto 50px auto;
  padding: 40px;

  background: ${props => props.theme.colors.neutralWhite};

  border-radius: 8px;
  box-shadow: 0px 16px 32px rgba(30, 42, 50, 0.38);

  @media only screen and (max-width: 380px) {
    border-radius: 4px;
    padding: 24px;
  }
`;
