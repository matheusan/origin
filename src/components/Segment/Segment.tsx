import * as React from 'react';

import * as S from './styled';

const Segment: React.SFC<SegmentProps> = ({ children }): JSX.Element => (
  <S.Segment>{children}</S.Segment>
);

export default Segment;
