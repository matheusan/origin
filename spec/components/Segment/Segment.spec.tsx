import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Segment from '../../../src/components/Segment';

describe('Segment', () => {
  let wrapper: ShallowWrapper;

  it('renders', () => {
    wrapper = shallow(<Segment>This is my content</Segment>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render all child elements', () => {
    wrapper = shallow(
      <Segment>
        <span>This is the first element</span>
        <div>This is the second element</div>
        <strong>This is a strong element</strong>
        <ul>
          <li>This</li>
          <li>is</li>
          <li>a</li>
          <li>multilevel</li>
          <li>element</li>
        </ul>
      </Segment>
    );
    expect(wrapper.find('span').length).toBe(1);
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.find('strong').length).toBe(1);
    expect(wrapper.find('ul').length).toBe(1);
    expect(wrapper.find('ul li').length).toBe(5);
  });
});
