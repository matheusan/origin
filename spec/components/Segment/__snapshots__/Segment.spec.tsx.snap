// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Segment renders 1`] = `
<styled.section>
  This is my content
</styled.section>
`;
